internal/util/util.go

```diff
@@ -34,6 +37,81 @@ func ValuesChecksum(values chartutil.Values) string {
 	return fmt.Sprintf("%x", sha1.Sum([]byte(s)))
 }
 
+// OrderedValuesChecksum sort the chartutil.Values then calculates
+// and returns the SHA1 checksum for the sorted values.
+func OrderedValuesChecksum(values chartutil.Values) string {
+	var s []byte
+	if len(values) != 0 {
+		// Check sum on the formatted values
+		newValues := copyValues(values)
+		msValues := yaml.JSONObjectToYAMLObject(newValues)
+		// Sort
+		SortMapSlice(msValues)
+		// Marshal
+		s, _ = goyaml.Marshal(msValues)
+	}
+	// Get hash
+	return fmt.Sprintf("%x", sha1.Sum(s))
+}
+
+func SortMapSlice(ms goyaml.MapSlice) {
+	sort.Slice(ms, func(i, j int) bool {
+		return fmt.Sprint(ms[i].Key) < fmt.Sprint(ms[j].Key)
+	})
+	for _, item := range ms {
+		if nestedMS, ok := item.Value.(goyaml.MapSlice); ok {
+			SortMapSlice(nestedMS)
+		} else if _, ok := item.Value.([]interface{}); ok {
+			for _, vItem := range item.Value.([]interface{}) {
+				if itemMS, ok := vItem.(goyaml.MapSlice); ok {
+					SortMapSlice(itemMS)
+				}
+			}
+		}
+	}
+}
+
+func cleanUpMapValue(v interface{}) interface{} {
+	switch v := v.(type) {
+	case []interface{}:
+		return cleanUpInterfaceArray(v)
+	case map[interface{}]interface{}:
+		return cleanUpInterfaceMap(v)
+	default:
+		return v
+	}
+}
+
+func cleanUpInterfaceMap(in map[interface{}]interface{}) map[string]interface{} {
+	result := make(map[string]interface{})
+	for k, v := range in {
+		result[fmt.Sprintf("%T.%v", k, k)] = cleanUpMapValue(v)
```

Comment:
Why are we taking note of the type here? It should take the checksum of an ordered map, without making any modifications to the map itself besides ordering.